recipes/node-url-to-whatwg-url/src/url-parse.ts

```diff
@@ -0,0 +1,128 @@
+
+    // Property access: obj.auth -> `${obj.username}:${obj.password}`
+    const authAccesses = rootNode.findAll({ rule: { pattern: "$OBJ.auth" } });
+    for (const node of authAccesses) {
+        const base = node.getMatch("OBJ");
+        if (!base) continue;
+
+        const replacement = `\`\${${base.text()}.username}:\${${base.text()}.password}\``;
+        edits.push(node.replace(replacement));
+    }
+
+    // Destructuring: const { auth } = obj -> const auth = `${obj.username}:${obj.password}`
+    const authDestructures = rootNode.findAll({ rule: { pattern: "const { auth } = $OBJ" } });
+    for (const node of authDestructures) {
+        const base = node.getMatch("OBJ");
+        if (!base) continue;
+        const hadSemi = /;\s*$/.test(node.text());
+        const name = base.text();
+        const replacement = `const auth = \`${'${'}${name}.username${'}'}:${'${'}${name}.password${'}'}\`${hadSemi ? ';' : ''}`;
+        edits.push(node.replace(replacement));
+    }
+
+    // Property access: obj.path -> `${obj.pathname}${obj.search}`
+    const pathAccesses = rootNode.findAll({ rule: { pattern: "$OBJ.path" } });
+    for (const node of pathAccesses) {
+        const base = node.getMatch("OBJ");
+        if (!base) continue;
+
+        const replacement = `\`\${${base.text()}.pathname}\${${base.text()}.search}\``;
+        edits.push(node.replace(replacement));
+    }
+
+    // Destructuring: const { path } = obj -> const path = `${obj.pathname}${obj.search}`
+    const pathDestructures = rootNode.findAll({ rule: { pattern: "const { path } = $OBJ" } });
+    for (const node of pathDestructures) {
+        const base = node.getMatch("OBJ");
+        if (!base) continue;
+        const hadSemi = /;\s*$/.test(node.text());
+        const name = base.text();
+        const replacement = `const path = \`${'${'}${name}.pathname${'}'}${'${'}${name}.search${'}'}\`${hadSemi ? ';' : ''}`;
+        edits.push(node.replace(replacement));
+    }
+
+    // Property access: obj.hostname -> obj.hostname.replace(/^\[|\]$/, '')
+    const hostnameAccesses = rootNode.findAll({ rule: { pattern: "$OBJ.hostname" } });
+    for (const node of hostnameAccesses) {
+        const base = node.getMatch("OBJ");
+        if (!base) continue;
+
+        const replacement = `${base.text()}.hostname.replace(/^\\[|\\]$/, '')`;
+        edits.push(node.replace(replacement));
+    }
+
+    // Destructuring: const { hostname } = obj -> const hostname = obj.hostname.replace(/^\[|\]$/, '')
+    const hostnameDestructures = rootNode.findAll({ rule: { pattern: "const { hostname } = $OBJ" } });
+    for (const node of hostnameDestructures) {
+        const base = node.getMatch("OBJ");
+        if (!base) continue;
+        const hadSemi = /;\s*$/.test(node.text());
+        const replacement = `const hostname = ${base.text()}.hostname.replace(/^\\[|\\]$/, '')${hadSemi ? ';' : ''}`;
+        edits.push(node.replace(replacement));
+    }
```

Comment:
I think the logic here can be reutilized between the properties.



When more than one property is destructured at the same time, this behaviour not will work, 

i.g:
`const { auth, path, hostname } = myURL`


something like

```js
const fieldsToReplace => [
  {
    key: 'auth',
    replaceFn: (base, hadSemi) => (`const auth = \`${'${'}${name}.username${'}'}:${'${'}${name}.password${'}'}\`${hadSemi ? ';' : ''}`)
  },
	{
		key: 'path',
    replaceFn: (base, hadSemi) => (const path = \`${'${'}${name}.pathname${'}'}${'${'}${name}.search${'}'}\`${hadSemi ? ';' : ''}`)
	}

]
```